### PR TITLE
[BRE-761] Allow for multiple stage command creation

### PR DESCRIPTION
**Pull request by ragurney**
screen shot 2018-07-02 at 12 45 26 pm

Based off of #2832 

/cc @zendesk/samson

### References
 - Jira link: [BRE-761](https://zendesk.atlassian.net/browse/BRE-761)

### CODE REVIEW

File: app/models/stage.rb [hunk 1]
```diff
@@ -159,30 +159,21 @@ def environments
...
   def command_ids
     stage_commands.sort_by(&:position).map(&:command_id)
   end
 
-  def commands
-    stage_commands.sort_by(&:position).map(&:command).compact
-  end
-
-  def command=(c)
+  # reconcile new commands and positioning from command checkboxes
```
Comment (grosser) [inline app/models/stage.rb]: `or textare`

File: app/models/stage.rb [hunk 2]
```diff
@@ -252,19 +243,25 @@ def validate_not_auto_deploying_without_buddy
...
   end
 
   def project_lock?(lock)
     lock.resource_type == "Project" && lock.resource_equal?(project)
   end
+
+  def find_or_create_stage_command(command_or_id, position)
+    command_or_id = command_or_id.to_s
+
+    if existing_command = command_or_id.match?(/\A\d+\z/) && Command.find_by(id: Integer(command_or_id))
```
Comment (grosser) [inline app/models/stage.rb]: n+1 ... only rely on `.detect`

Comment (grosser) [reply app/models/stage.rb]: if you want to make sure it exists, could do `ids = Command.pluck(:id)` ... or some other way of making sure typing an integer does not mess it up :D

File: app/models/stage.rb [hunk 3]
```diff
@@ -252,19 +243,25 @@ def validate_not_auto_deploying_without_buddy
...
+      if stage_command = stage_commands.detect { |sc| sc.command_id == existing_command.id }
+        stage_command
+      else
+        StageCommand.new(command: existing_command)
```
Comment (grosser) [inline app/models/stage.rb]: did not assign project ?

File: app/views/stages/_commands.html.erb [hunk 4]
```diff
@@ -10,38 +10,49 @@
...
+</fieldset>
 
-  <hr />
+<hr/>
 
-  <p>Add a command which is specific to your project</p>
-  <%= form.text_area :command, class: "form-control", rows: 4, value: nil, placeholder: 'cap production deploy' %>
-</fieldset>
+<%= button_tag(
+        'Add Stage Command',
```
Comment (grosser) [inline app/views/stages/_commands.html.erb]: strange indent

File: app/models/stage.rb (hunk 3 above)
Comment (ragurney) [reply app/models/stage.rb]: I think if the command already exists then it should either have the correct `project_id` already assigned to it, or it's a global command and so shouldn't have a `project_id` assigned. Is that right?

**Comment by ragurney** [comment]
I don't think `StageCommand` has a `project_id` field.. just a `stage_id`.

**Comment by grosser** [comment]
ah right!

**Pull request merged**